lib/date_range.rb

```diff
@@ -0,0 +1,21 @@
+require 'time'
+
+class DateRange
+  # Check if given date is valid 
+  def self.valid_date?(date)
+    begin
```

Comment:
I like that you've isolated the logic for working with date ranges in this class - good organization.

I am curious about your decision to make these all self methods, rather than creating instances of a `DateRange` class. For an alternative way to approach it, you might check out the [instructor implementation](https://github.com/droberts-ada/hotel/blob/dpr/solution/lib/date_range.rb).